test/e2e/dhcp_overrides_test.go

```diff
@@ -0,0 +1,121 @@
+limitations under the License.
+*/
+
+package e2e
+
+import (
+	"encoding/base64"
+	"fmt"
+	"path/filepath"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/vmware/govmomi/find"
+	"github.com/vmware/govmomi/property"
+	"github.com/vmware/govmomi/vim25/mo"
+	"gopkg.in/yaml.v2"
+	"k8s.io/utils/pointer"
+	"sigs.k8s.io/cluster-api/test/framework/clusterctl"
+	"sigs.k8s.io/cluster-api/util"
+
+	types "sigs.k8s.io/cluster-api-provider-vsphere/apis/v1beta1"
+)
+
+type GuestInfoMetadata struct {
+	Network Network `yaml:"network"`
+}
+
+type Network struct {
+	Ethernets map[string]Ethernet `yaml:"ethernets"`
+}
+
+type Ethernet struct {
+	DHCP4Overrides DHCPOverrides `yaml:"dhcp4-overrides"`
+}
+
+type DHCPOverrides struct {
+	SendHostname *bool `yaml:"send-hostname"`
+}
+
+var _ = Describe("DHCPOverrides configuration test", func() {
+	When("Creating a cluster with DHCPOverrides configured", func() {
+		It("Only configures the network with the provided nameservers", func() {
+			specName := "dhcp-overrides"
+			namespace := setupSpecNamespace(specName)
+			clusterName := fmt.Sprintf("%s-%s", specName, util.RandomString(6))
+			clusterResources := new(clusterctl.ApplyClusterTemplateAndWaitResult)
+
+			By("Creating a workload cluster")
+			clusterctl.ApplyClusterTemplateAndWait(ctx, clusterctl.ApplyClusterTemplateAndWaitInput{
+				ClusterProxy: bootstrapClusterProxy,
+				ConfigCluster: clusterctl.ConfigClusterInput{
+					LogFolder:                filepath.Join(artifactFolder, "clusters", bootstrapClusterProxy.GetName()),
+					ClusterctlConfigPath:     clusterctlConfigPath,
+					KubeconfigPath:           bootstrapClusterProxy.GetKubeconfigPath(),
+					InfrastructureProvider:   clusterctl.DefaultInfrastructureProvider,
+					Flavor:                   "dhcp-overrides",
+					Namespace:                namespace.Name,
+					ClusterName:              clusterName,
+					KubernetesVersion:        e2eConfig.GetVariable(KubernetesVersion),
+					ControlPlaneMachineCount: pointer.Int64Ptr(1),
+					WorkerMachineCount:       pointer.Int64Ptr(1),
+				},
+				WaitForClusterIntervals:      e2eConfig.GetIntervals(specName, "wait-cluster"),
+				WaitForControlPlaneIntervals: e2eConfig.GetIntervals(specName, "wait-control-plane"),
+				WaitForMachineDeployments:    e2eConfig.GetIntervals(specName, "wait-worker-nodes"),
+			}, clusterResources)
+
+			list := getVSphereVMsForCluster(clusterName, namespace.Name)
+			for _, vm := range list.Items {
+				metadata := getVMMetadata(vm)
+				guestInfoMetadata := &GuestInfoMetadata{}
+				err := yaml.Unmarshal(metadata, guestInfoMetadata)
+				Expect(err).NotTo(HaveOccurred())
+				Expect(guestInfoMetadata.Network.Ethernets["id0"].DHCP4Overrides.SendHostname).NotTo(BeNil())
+				Expect(*guestInfoMetadata.Network.Ethernets["id0"].DHCP4Overrides.SendHostname).To(BeFalse())
+			}
+		})
+	})
+})
+
+func getVMMetadata(vm types.VSphereVM) []byte {
```

Comment:
rewrite the signature to send an error if the metadata is not found and assert on the error on line 83, where the function is called.

Reply:
Updated, thanks!